src/api.v2/model/experiment.js

```diff
@@ -1,22 +1,91 @@
+const _ = require('lodash');
+
+/* eslint-disable func-names */
 const generateBasicModelFunctions = require('../helpers/generateBasicModelFunctions');
+const sqlClient = require('../../sql/sqlClient');
+const { aggregateIntoJson } = require('../../sql/helpers');
+
+const getLogger = require('../../utils/getLogger');
+
+const logger = getLogger('[ExperimentModel] - ');
 
-const tableName = 'experiment';
+const experimentTable = 'experiment';
+const experimentExecutionTable = 'experiment_execution';
 
-const selectableProps = [
+const experimentFields = [
   'id',
   'name',
   'description',
-  'processing_config',
+  'samples_order',
   'notify_by_email',
   'created_at',
   'updated_at',
 ];
 
 const basicModelFunctions = generateBasicModelFunctions({
-  tableName,
-  selectableProps,
+  tableName: experimentTable,
+  selectableProps: experimentFields,
 });
 
+const getExperimentData = async (experimentId) => {
+  const sql = sqlClient.get();
+
+  function query() {
+    this.select('*')
+      .from(experimentTable)
+      .leftJoin(experimentExecutionTable, `${experimentTable}.id`, `${experimentExecutionTable}.experiment_id`)
+      .where('id', experimentId)
+      .as('experiment_with_exec');
+  }
+
+  const experimentExecutionFields = [
+    'params_hash', 'state_machine_arn', 'execution_arn',
+  ];
+
+  const result = await aggregateIntoJson(query, experimentFields, experimentExecutionFields, 'pipeline_type', 'pipelines', sql);
+
+  return result[0];
+};
+
+const updateSamplePosition = async (experimentId, oldPosition, newPosition) => {
+  const sql = sqlClient.get();
+
+  // Sets samples_order as an array that has the sample in oldPosition moved to newPosition
+
+  const trx = await sql.transaction();
+
+  try {
+    const result = await trx(experimentTable).update({
+      samples_order: trx.raw(`(
+        SELECT jsonb_insert(samples_order - ${oldPosition}, '{${newPosition}}', samples_order -> ${oldPosition}, false)
```

Comment:
in `jsonb_insert`?

It can work for both objects and arrays, but since in this case we are working with a jsonb array (`samples_order`) I'll explain it as an array.

It takes 3/4 parameters (last bool param is optional) and returns a jsonb array with a new element inserted.

1st param: original array
2nd param: (format: `'{5}'`), the index at which you want to insert the value.
3rd param: The value that we want to insert.
4th param: "Insert after", determines whether the new value is inserted before or after the element that already exists at `newPosition`.